### PR TITLE
feat(appconfig): support for CfnMonitorsProperty in environments

### PR DESCRIPTION
**Pull request by chenjane-dev**
Adding support to associate our L1 monitors constructs (CfnMonitorsProperty) to a L2 environment construct.

**BREAKING CHANGE**
To define monitors under an environment, you now need to call a static method on `Monitor`.
Example:

```
new Environment(this, 'MyEnv', {
   ....
   monitors: [
      Monitor.fromCloudWatchAlarm(...),
      Monitor.fromCfnMonitorProperty(...),
   ],
});
```

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-appconfig-alpha/lib/environment.ts [hunk 1]
```diff
@@ -276,21 +284,57 @@ export class Environment extends EnvironmentBase {
...
+
+  /**
+   * The alarm ARN for AWS AppConfig to monitor.
+   */
+  public abstract readonly alarmArn: string;
+
+  /**
+   * The IAM role ARN for AWS AppConfig to view the alarm state.
+   */
+  public abstract readonly alarmRoleArn?: string;
```
Comment (scanlonp) [inline packages/@aws-cdk/aws-appconfig-alpha/lib/environment.ts:332]: nit: this optional class variable is defined before a required one.

File: packages/@aws-cdk/aws-appconfig-alpha/lib/extension.ts [hunk 2]
```diff
@@ -290,21 +290,29 @@ export interface ExtensionAttributes {
 
   /**
    * The Amazon Resource Name (ARN) of the extension.
+   *
+   * @default - The ARN is generated.
```
Comment (scanlonp) [inline packages/@aws-cdk/aws-appconfig-alpha/lib/extension.ts:294]: nit: wish this was a little more descriptive.

File: packages/@aws-cdk/aws-appconfig-alpha/README.md [hunk 3]
```diff
@@ -330,8 +330,8 @@ declare const alarm: cloudwatch.Alarm;
 new appconfig.Environment(this, 'MyEnvironment', {
   application,
   monitors: [
-    {alarm},
-  ]
+    appconfig.Monitor.fromCloudWatchAlarm(alarm),
+  ],
 });
 ```
```
Comment (scanlonp) [inline packages/@aws-cdk/aws-appconfig-alpha/README.md:337]: We should have an example for `fromCfnMonitorsProperty` unless it is not the recommended approach. Even then the readme could have a line saying that it is an option but not the recommended approach

**Review by scanlonp — DISMISSED** [review]
Looks good, thank you!
A couple very minor nits and we should be good to go.

File: packages/@aws-cdk/aws-appconfig-alpha/README.md (hunk 3 above)
Comment (chenjane-dev) [reply packages/@aws-cdk/aws-appconfig-alpha/README.md:337]: Updated

File: packages/@aws-cdk/aws-appconfig-alpha/lib/environment.ts (hunk 1 above)
Comment (chenjane-dev) [reply packages/@aws-cdk/aws-appconfig-alpha/lib/environment.ts]: Updated

File: packages/@aws-cdk/aws-appconfig-alpha/lib/extension.ts (hunk 2 above)
Comment (chenjane-dev) [reply packages/@aws-cdk/aws-appconfig-alpha/lib/extension.ts]: Changed to `The extension ARN is generated`

**Review by scanlonp — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**